tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Remoting/Client/HttpProcessMessageIntegration.cs

```diff
@@ -0,0 +1,97 @@
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { "System.Runtime.Remoting.Messaging.IMessage", "System.Runtime.Remoting.Channels.ITransportHeaders", "System.IO.Stream", "System.Runtime.Remoting.Channels.ITransportHeaders&", "System.IO.Stream&", },
+        MinimumVersion = RemotingIntegration.Major4,
+        MaximumVersion = RemotingIntegration.Major4,
+        IntegrationName = RemotingIntegration.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    // ReSharper disable once InconsistentNaming
+    public class HttpProcessMessageIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="msg">The incoming request message instance</param>
+        /// <param name="requestHeaders">The headers for the incoming request message</param>
+        /// <param name="requestStream">The stream for the incoming request message</param>
+        /// <param name="responseHeaders">The headers for the outgoing response message</param>
+        /// <param name="responseStream">The stream for the outgoing response message</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, IMessage msg, ITransportHeaders requestHeaders, Stream requestStream, ref ITransportHeaders responseHeaders, ref Stream responseStream)
+        {
+            // Create a new trace_id/span_id for the child WebRequest span
+            var tracer = Tracer.Instance;
+            if (tracer.Settings.IsIntegrationEnabled(WebRequestCommon.IntegrationId))
+            {
+                // If we create a scope, we can update the following attributes in other automatic instrumentation methods:
+                // - Method
+                // - Uri
+                // - ResourceName (Recalculate with both Method and Uri)
+
+                Scope scope = ScopeFactory.CreateOutboundHttpScope(tracer, httpMethod: null, requestUri: null, WebRequestCommon.IntegrationId, out _);
+                if (scope != null)
+                {
+                    // Add distributed tracing headers to the HTTP request.
+                    // The expected sequence of calls is GetRequestStream -> GetResponse. Headers can't be modified after calling GetRequestStream.
+                    // At the same time, we don't want to set an active scope now, because it's possible that GetResponse will never be called.
+                    // Instead, we generate a spancontext and inject it in the headers. GetResponse will fetch them and create an active scope with the right id.
+                    SpanContextPropagator.Instance.Inject(scope.Span.Context, requestHeaders, (headers, key, value) => headers[key] = value);
```

Comment:
No action needed, but I just remembered something: We recently started adding a call to `Datadog.Trace.ClrProfiler.AutoInstrumentation.Http.WebRequest.HeadersInjectedCache.SetInjectedHeaders(request.Headers)` when `WebRequest.GetRequestStream` is called. I don't think that will be necessary here because:
- We disable any of our WebRequest integration from firing with the `requestHeaders["x-datadog-tracing-enabled"] = "false";` below
- The user isn't updating the request headers manually, it is the remoting infrastructure that is doing all the work

In case we ever get weird distributed tracing headers, we know where to look. But I don't think it will be an issue